### PR TITLE
Update 50 - Cucumber Tests

### PR DESCRIPTION
**Pull request by stringhandler**

### CODE REVIEW

File: _updates/2021-03-15-update-50.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+author: Mike Berry
+thumbnail: update-background.jpg
+title: Behavior Driven Developemnt (BDD) Testing with Cucumber
+subtitle:
+class: subpage
+---
+
+## Behavior Driven Development (BDD) Testing with Cucumber
+
+You may have noticed a lot of PR's and commits with the word "cucumber" in them. Tari uses [Cucumber](https://cucumber.io/), and specifically [Cucumber.js](https://github.com/cucumber/cucumber-js) to easily specify test scenarios in plain text. The tests can be found under the [integration_tests/features](https://github.com/tari-project/tari/tree/development/integration_tests/features) folder. The tests spin up base nodes, merge mining proxies and miners in a local, low difficulty network and interact with them via GRPC, simulating various scenarios that might happen on testnet.
```
Comment (hansieodendaal) [inline _updates/2021-03-15-update-50.md:17]: **NIT:**
Maybe include "integration tests" here:
```
... and commits with the words "cucumber" or "integration tests" in ...
```
vs.
```
... and commits with the word "cucumber" in ...
```

File: _updates/2021-03-15-update-50.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    When I mine a block on SEED_A with coinbase CB1
+    And I mine 2 blocks on SEED_A
+    And I create a transaction TX1 spending CB1 to UTX1
+    And I submit transaction TX1 to SEED_A
+    Then TX1 is in the mempool
+    When I mine 1 blocks on SEED_A
+    Then the UTXO UTX1 has been mined according to SEED_A
+
+```
+
```
Comment (hansieodendaal) [inline _updates/2021-03-15-update-50.md:64]: **NIT:**
```suggestion
We can even use Cucumber to [send and receive thousands of transactions](https://github.com/tari-project/tari/blob/development/integration_tests/features/StressTest.feature) within this controlled environment.

```

File: _updates/2021-03-15-update-50.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+tag: Developer Update
+date: 2021-03-15
+author: Mike Berry
+thumbnail: update-background.jpg
+title: Behavior Driven Developemnt (BDD) Testing with Cucumber
+subtitle:
+class: subpage
+---
+
+## Behavior Driven Development (BDD) Testing with Cucumber
```
Comment (CjS77) [inline _updates/2021-03-15-update-50.md]:
```suggestion
## Tari is going green!
No, we're not switching to Proof of Stake. (Lol - [as if that made a difference](https://www.truthcoin.info/blog/pow-cheapest).

But we are using Cucumber for Behavior Driven Development (BDD).
```

File: _updates/2021-03-15-update-50.md (hunk 2 above)
Comment (CjS77) [reply _updates/2021-03-15-update-50.md:64]: I think the basic example is sufficient for now

**Review by CjS77 — APPROVED** [review]
Much better draft. Let's ship it!

**Pull request merged**